AssemblyScript/assemblyscript
# Update Binaryen

Update Binaryen to latest, including:

* Rename `dataref` to `structref` and update subtyping accordingly (fixes #2551)
* Consolidate `ref.is_*` to `ref.is_null`. `ref.is_func`, `.is_data` and `.is_i31` are now expressible with `ref.test`.
* `ref.as_func`, `.as_data` and `.as_i31` are now expressible with `ref.cast`.
* `br_on_func`, `_on_non_func`, `_on_data`, `_on_non_data`, `_on_i31` and `_on_non_i31` are now expressible with `br_on_cast`.
* Remove explicit type on `ref.cast`.
* Apply related renamings, i.e. `intendedType` to `castType`.

Has some overlap with #2547 but does not update the pass pipeline.

- [x] I've added my name and email to the NOTICE file
